stable_baselines3/common/save_util.py

```diff
@@ -255,7 +260,7 @@ def open_path_pathlib(path: pathlib.Path, mode: str, verbose: int = 0, suffix: O
 
     if mode == "r":
         try:
-            path = path.open("rb")
+            path = path.open("rb")  # type: ignore[assignment]
```

Comment:
Oh ok. It took me a while to get it and at first it looked like line 291 would lead to an infinite recursion in the "happy path". Only then I scrolled up and realized that the type assumptions are broken and this breaks the infinite recursion. I think it might have been easier to understand with two calls to to `open_path`. Or maybe even call `open_path` here and explicitly call `open_path_pathlib` below to make it even more clear that the below calls are "reattempts with better/corrected arguments.

Reply:
should be fixed in https://github.com/DLR-RM/stable-baselines3/pull/1648/commits/ee2453896d26268b9c551f401ef23a92be2ae822